python/mxnet/ndarray/ndarray.py

```diff
@@ -2086,12 +2088,17 @@ def copyto(self, other):
             if other.handle is self.handle:
                 warnings.warn('You are attempting to copy an array to itself', RuntimeWarning)
                 return False
-            return _internal._copyto(self, out=other)
+            data = _internal._copyto(self, out=other)
         elif isinstance(other, Context):
             hret = NDArray(_new_alloc_handle(self.shape, other, True, self.dtype))
-            return _internal._copyto(self, out=hret)
+            data = _internal._copyto(self, out=hret)
         else:
             raise TypeError('copyto does not support type ' + str(type(other)))
+        if copy_grad:
+            grad = self.grad
+            if grad is not None:
+                setattr(data, '_old_grad', grad.copyto(other, copy_grad=False))
```

Comment:
This seems rather obscure as users aren't aware that this field exists. If user explicitly chose to do the copy, I think it's ok to set it in `.grad` attribute

Reply:
What's less clear to me is if copy_grad is true, the new grad array should be marked as the gradient of the new data array in the target context. Again, this seems to implicit.

Reply:
There is a case:
```python
a = mx.nd.array([1,2,3],ctx=mx.cpu())
a.attach_grad()
b = a.as_in_context()
print(b.grad) # return a gradient which is equal to a.grad
b.attach_grad()
print(b.grad) # return a new gradient
```
I need the attribute `_old_grad` to store the gradient.